### PR TITLE
refactor(nix): cleanup and switch to `nixfmt-rfc-style`

### PR DESCRIPTION
**Pull request by uncenter**
Cleans up the code with the `statix` and `deadnix` tools, and switches the formatter to `nixfmt-rfc-style`.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -49,9 +49,9 @@
     // {
       overlays = rec {
         default = yazi;
-        yazi = final: prev: {
-          yazi-unwrapped = self.packages."${final.system}".yazi-unwrapped;
-          yazi = self.packages."${final.system}".yazi;
+        yazi = final: _: {
+          inherit (self.packages."${final.system}") yazi-unwrapped;
+          inherit (self.packages."${final.system}") yazi;
```
Comment (isabelroses) [inline flake.nix:54]:
```suggestion
          inherit (self.packages."${final.system}") yazi yazi-unwrapped;
```

File: nix/yazi.nix [hunk 2]
```diff
@@ -24,8 +24,7 @@
 }:
 
 let
-  runtimePaths = with lib; [ ]
-    ++ optional withFile file
+  runtimePaths = with lib; optional withFile file
```
Comment (isabelroses) [inline nix/yazi.nix:27]: you can also remove this `with lib;` by doing `inherit (lib) optional makeBinPath;`

**Comment by sxyazi** [comment]
I believe this is a meaningful change, but my limited Nix knowledge isn't enough for me to review it right now.

@XYenon or @linsui, could you please have a look at this PR?

**Comment by uncenter** [comment]
> I believe this is a meaningful change, but my limited Nix knowledge isn't enough for me to review it right now.
> 
> @XYenon or @linsui, could you please have a look at this PR?

This PR doesn't change anything about the build, just the formatting and cleans up useless ("dead") code.

**Review by XYenon — APPROVED** [review]
LGTM :)

**Review by sxyazi — APPROVED** [review]
Great, thanks to @uncenter for the refactoring and @XYenon, @isabelroses for the review! 

Let's go ahead and merge now.

**Comment by sxyazi** [comment]
Oh one more thing, the dependency list for Yazi 0.3 has a bit changed, https://github.com/sxyazi/yazi/issues/1046#issuecomment-2255993749. 

@uncenter would you like to raise an extra PR for it?

**Comment by uncenter** [comment]
I'll look into it! Usually the Nixpkgs folks are more experienced with that though so I might see what changes they make and copy them over here.

**Comment by XYenon** [comment]
> Oh one more thing, the dependency list for Yazi 0.3 has a bit changed, [#1046 (comment)](https://github.com/sxyazi/yazi/issues/1046#issuecomment-2255993749).
> 
> @uncenter would you like to raise an extra PR for it?

I'll make a PR for it after #1380 be merged.

**Comment by uncenter** [comment]
> I'll make a PR for it after #1380 be merged.

We should also update the way optional dependencies are provided as done in Nixpkgs I think, see https://github.com/NixOS/nixpkgs/commit/1f6b8d25b8b364908ca2aeac3bfc4b9ebb7d6aa4.

**Comment by uncenter** [comment]
> > Oh one more thing, the dependency list for Yazi 0.3 has a bit changed, [#1046 (comment)](https://github.com/sxyazi/yazi/issues/1046#issuecomment-2255993749).
> > @uncenter would you like to raise an extra PR for it?
> 
> I'll make a PR for it after #1380 be merged.

I've made an upstream Nixpkgs update PR, https://github.com/NixOS/nixpkgs/pull/331563. Assuming everything is correct we should just be able to copy that over here after #1380 :)

**Pull request merged**